File: backend/src/main/scala/order/api/ResolvePartialRefundRequestApi.scala
package order.api

import domain.shared.given

import admin.app.*
import cats.effect.IO
import domain.order.ResolvePartialRefundRequest
import domain.shared.{DeliveryAppState, RefundRequestId, UserRole}
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import org.http4s.dsl.io.*
import shared.api.routing.*
import shared.app.*

val resolvePartialRefundRequestApi: FixedMethodApi1[RefundRequestId, DeliveryAppState] =
  jsonPostApi1[RefundRequestId, ResolvePartialRefundRequest, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("partial-refunds")),
    List(routeSegment("review")),
  )

File: frontend/src/apis/order/ResolvePartialRefundRequestApi.ts
import type { DeliveryAppState, RefundRequestId, ResolvePartialRefundRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { buildApiPath1, defineJsonPostApi1, routeSegment } from '@/system/api/TypedApiDefinitions'

export const resolvePartialRefundRequestApiDefinition = defineJsonPostApi1<RefundRequestId, ResolvePartialRefundRequest, DeliveryAppState>(
    [routeSegment('api'), routeSegment('delivery'), routeSegment('partial-refunds')],
    [routeSegment('review')],
  )

export function resolvePartialRefundRequest(
  refundId: RefundRequestId,
  payload: ResolvePartialRefundRequest,
) {
  return postNormalizedDeliveryState(buildApiPath1(resolvePartialRefundRequestApiDefinition, refundId), payload)
}
